src/sysfiles.c

```diff
@@ -166,12 +166,12 @@ static ssize_t echoandcheck(int fid, const char *buf, size_t count)
   else {
       ret = write(syBuf[fid].echo, buf, count);
       if (ret < 0) {
-          if (syBuf[fid].fp == fileno(stdout) || syBuf[fid].fp == fileno(stderr)) {
+          if (syBuf[fid].echo == fileno(stdout) || syBuf[fid].echo == fileno(stderr)) {
               Panic("Could not write to stdout/stderr.");
```

Comment:
We might want to give a bit more information here and also give the value of `fid`? Same below... @ChrisJefferson ?

Reply:
Yes, I think knowing 'fid' would be useful, in the unlikely event people see this message it will give us a better idea what has gone wrong.